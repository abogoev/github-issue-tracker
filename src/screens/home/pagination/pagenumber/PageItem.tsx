import React, { VFC } from "react";
import { StyleSheet, Text, TouchableOpacityProps } from "react-native";
import theme from "../../../../theme/theme";
import CustomTouchableOpacity from "../../../../core/touchableopacity/CustomTouchableOpacity";

export interface Props extends TouchableOpacityProps {
  text: string;
  isActive?: boolean;
}

const PageItem: VFC<Props> = ({ text, isActive, style, ...props }) => {
  return (
    <CustomTouchableOpacity
      accessibilityRole="button"
      style={[styles(isActive).container, style]}
      {...props}
    >
      <Text style={styles(isActive).text}>{text}</Text>
    </CustomTouchableOpacity>
  );
};

const styles = (isActive?: boolean) =>
  StyleSheet.create({
    container: {
      alignSelf: "flex-start",
      paddingHorizontal: 10,
      paddingVertical: 5,
      borderRadius: theme.borderRadius,
      backgroundColor: isActive
        ? theme.palette.blue
        : theme.palette.transparent,
      borderColor: isActive ? theme.palette.transparent : theme.palette.neutral,
      borderWidth: 1,
    },
    text: {
      ...theme.typography.regularTextBold,
      color: isActive ? theme.palette.white : theme.palette.veryDarkGrey,
    },
  });

export default PageItem;
