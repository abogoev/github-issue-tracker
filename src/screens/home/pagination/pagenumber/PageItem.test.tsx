import React from "react";
import PageItem, { Props } from "./PageItem";
import { create } from "react-test-renderer";

describe("PageItem", () => {
  it("should render correctly", () => {
    const props: Props = { text: "1" };
    let component = create(<PageItem {...props} />);

    expect(component.toJSON()).toMatchSnapshot("when page is not active");

    props.isActive = true;
    component = create(<PageItem {...props} />);

    expect(component.toJSON()).toMatchSnapshot("when page is active");
  });
});
