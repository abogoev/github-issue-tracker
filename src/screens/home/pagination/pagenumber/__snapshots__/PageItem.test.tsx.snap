// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageItem should render correctly: when page is active 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "backgroundColor": "rgb(9, 105, 218)",
      "borderColor": "transparent",
      "borderRadius": 6,
      "borderWidth": 1,
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Helvetica",
        "fontSize": 14,
        "fontWeight": "500",
        "lineHeight": 20,
      }
    }
  >
    1
  </Text>
</View>
`;

exports[`PageItem should render correctly: when page is not active 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderColor": "rgb(208, 215, 222)",
      "borderRadius": 6,
      "borderWidth": 1,
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#24292f",
        "fontFamily": "Helvetica",
        "fontSize": 14,
        "fontWeight": "500",
        "lineHeight": 20,
      }
    }
  >
    1
  </Text>
</View>
`;
